src/tcache.c

```diff
@@ -747,30 +753,34 @@ tcache_boot(tsdn_t *tsdn) {
 	nhbins = sz_size2index(tcache_maxclass) + 1;
 
 	/* Initialize tcache_bin_info. */
-	tcache_bin_info = (cache_bin_info_t *)base_alloc(tsdn, b0get(), nhbins
-	    * sizeof(cache_bin_info_t), CACHELINE);
-	if (tcache_bin_info == NULL) {
-		return true;
+	assert(nhbins >= SC_NBINS);
+	if (nhbins > SC_NBINS) {
+		tcache_large_bin_info = (cache_bin_info_t *)base_alloc(tsdn,
+		    b0get(), (nhbins - SC_NBINS) * sizeof(cache_bin_info_t),
+		    CACHELINE);
+		if (tcache_large_bin_info == NULL) {
+			return true;
+		}
```

Comment:
nit: let's move this section to below the first loop, to make it clear that this is allocating for the large sizes.

Also there you could simply say, `if (nhbins == SC_NBINS) return;`